### PR TITLE
Update rails 7.1.1

### PR DESCRIPTION
**Pull request by chao-xian**
Here are the steps I did:

1. Bumped Rails in the Gemfile
2. bundled
3. ran `THOR_MERGE="code -d $1 $2" be bin/rails app:update`
4. for each change chose the `m` option to go through the changes, comparing against https://railsdiff.org/7.0.8/7.1.1
5. at the same time had a Guardfile running `bundle exec rake` to run the whole test suite on any file changes

The config defaults were already flipped to 7.1 in `application.rb` as part of the update above.

Also ran up the app with `rails s` and all looked fine.

### Things to consider when reviewing

- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -16,13 +16,12 @@
...
-  # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
+  # Ensures that a master key has been made available in ENV["RAILS_MASTER_KEY"], config/master.key, or an environment
+  # key such as config/credentials/production.key. This key is used to decrypt credentials (and other encrypted files).
   # config.require_master_key = true
 
-  # Disable serving static files from the `/public` folder by default since
-  # Apache or NGINX already handles this.
-  config.public_file_server.enabled = ENV["RAILS_SERVE_STATIC_FILES"].present?
+  # Enable static file serving from the `/public` folder (turn off if using NGINX/Apache for it).
+  config.public_file_server.enabled = true
```
Comment (aliuk2012) [inline config/environments/production.rb:24]: Are we not using `RAILS_SERVE_STATIC_FILES` env var in forms-deploy? if not  then we should remove it

File: config/environments/production.rb [hunk 2]
```diff
@@ -31,16 +30,26 @@
   # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for Apache
   # config.action_dispatch.x_sendfile_header = "X-Accel-Redirect" # for NGINX
 
+  # Assume all access to the app is happening through a SSL-terminating reverse proxy.
+  # Can be used together with config.force_ssl for Strict-Transport-Security and secure cookies.
+  # config.assume_ssl = true
```
Comment (aliuk2012) [inline config/environments/production.rb:35]: 🤔  should this not be uncommented?

File: config/environments/production.rb [hunk 3]
```diff
@@ -31,16 +30,26 @@
...
+    .tap  { |logger| logger.formatter = ::Logger::Formatter.new }
+    .then { |logger| ActiveSupport::TaggedLogging.new(logger) }
 
   # Prepend all log lines with the following tags.
   config.log_tags = [:request_id]
 
+  # Info include generic and useful information about system operation, but avoids logging too much
+  # information to avoid inadvertent exposure of personally identifiable information (PII). If you
+  # want to log everything, set the level to "debug".
+  config.log_level = ENV.fetch("RAILS_LOG_LEVEL", "info")
```
Comment (aliuk2012) [inline config/environments/production.rb:51]: Do we have `RAILS_LOG_LEVEL` env var setup in forms-deploy?  if we going to use env vars to set this, could I suggest we use config gem settings.yaml to configure this value.?

File: config/environments/production.rb [hunk 4]
```diff
@@ -51,16 +60,11 @@
...
   config.active_support.report_deprecations = false
 
-  # Use default logging formatter so that PID and timestamp are not suppressed.
-  config.log_formatter = ::Logger::Formatter.new
-
-  # Use a different logger for distributed setups.
-  # require "syslog/logger"
-  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new "app-name")
-
-  if ENV["RAILS_LOG_TO_STDOUT"].present?
```
Comment (aliuk2012) [inline config/environments/production.rb:61]: Are we not using RAILS_LOG_TO_STDOUT env var in forms-deploy? if not then we should remove it

File: config/environments/production.rb (hunk 3 above)
Comment (chao-xian) [reply config/environments/production.rb:51]: The value before was hard coded to `:info`. I think here the default is still that. This change is incoming from the upgrade.

File: config/environments/production.rb (hunk 4 above)
Comment (chao-xian) [reply config/environments/production.rb:61]: [Seems that as 7.1 sets it to output to stdout by default](https://github.com/alphagov/forms-product-page/pull/88/files/527e086ac9780ed208ac73ff343693957d989ae1..a422830e0658f61f6ea643a83c7a846a4a649dc3#diff-da60b4e96eff2b132991226d308949e23f4ef3aad45ad59edd09cbc32cc6251eR40) so I'll remove from forms-deploy

Comment (chao-xian) [reply config/environments/production.rb:61]: https://github.com/alphagov/forms-deploy/pull/395

File: config/environments/production.rb (hunk 1 above)
Comment (chao-xian) [reply config/environments/production.rb]: Ah damn I thought we should keep it but in fact we should remove from forms-deploy... I'll add that to the other PR and bring it back here...

**Review by aliuk2012 — APPROVED** [review]

File: config/environments/production.rb (hunk 2 above)
Comment (chao-xian) [reply config/environments/production.rb]: Oops

File: config/environments/production.rb (hunk 1 above)
Comment (chao-xian) [reply config/environments/production.rb]: On second thought it's an incoming change and we don't already have this set in forms-deploy so leaving it as the default. We are currently serving a commented out robots.txt but I do think we should actually set something there...

**Pull request merged**